gninasrc/lib/flexinfo.cpp

```diff
@@ -115,9 +192,15 @@ void FlexInfo::extractFlex(OpenBabel::OBMol& receptor, OpenBabel::OBMol& rigid,
   b.expand(flex_dist);
   double flsq = flex_dist * flex_dist;
 
+  std::vector<std::size_t> residues_idxs;
   FOR_ATOMS_OF_MOL(a, rigid){
-    if(a->GetAtomicNum() == 1)
-    continue; //heavy atoms only
+    if(a->GetAtomicNum() == 1) continue; //heavy atoms only
+    if (a->GetResidue() != NULL){ // Check
+      // TODO: @RMeli figure out exactly then this happens 
+      if (!isSideChain(a->GetResidue()->GetAtomID(&(*a))))
```

Comment:
This causes a segmentation fault for the tests where an `.xyz` file is used (such as `grid2gpu`). The reason is that `a->GetResidue()` returns a `NULL` pointer in that situation (as already pointed out by @dkoes ). I feel that `if (a->GetResidue() != NULL)` is an hacky solution, do you have alternative suggestions? 

I think skipping the call `FlexInfo::extractFlex` in `MolGetter::create_init_model` could be better, but I'm not sure I fully understand all the code branches of  such function. Any advice?

Reply:
I left `if (a->GetResidue() != NULL)` as a guard, but added an if/else block in `MolGetter::create_init_model` to call `FlexInfo::extractFlex` only with PDB files. I feel this is a better workaround.